test/functional/readpreference.test.js

```diff
@@ -548,4 +550,102 @@ describe('ReadPreference', function() {
       client.close(done);
     });
   });
+
+  context('hedge', function() {
+    before(function() {
+      this.testHedge = callback => {
+        const { configuration } = this;
+        const client = configuration.newClient(configuration.writeConcernMax(), {
+          poolSize: 1,
+          monitorCommands: true
+        });
+        const events = [];
+        client.on('commandStarted', filterForCommands(['find'], events));
+        client.connect(function(err, client) {
+          test.equal(null, err);
+          const collection = client.db(configuration.db).collection('test');
+          return callback(client, collection, events);
+        });
+      };
+    });
+
+    const itWithMeta = (a, b) => {
+      it(a, {
+        metadata: { requires: { topology: ['sharded'] } },
+        test: b
+      });
+    };
+
+    itWithMeta('should set hedge using [find option & empty hedge]', function(done) {
+      const rp = new ReadPreference(ReadPreference.SECONDARY, null, { hedge: {} });
+      this.testHedge((client, collection, events) => {
+        collection.find({}, { readPreference: rp }).toArray(err => {
+          test.equal(null, err);
+          const expected = { mode: ReadPreference.SECONDARY, hedge: {} };
+          expect(events[0].command.$readPreference).to.deep.equal(expected);
```

Comment:
For these expectations, I think something like:
`expect(events[0]).nested.property('command.$readPreference').to.deep.equal(expected);`
would be more idiomatic.